.github/CONTRIBUTING.md

```diff
@@ -21,39 +21,64 @@ Open an issue on our [issues list](https://bugs.openjdk.java.net) on JBS which d
 
-### Submitting your changes
+### Before submitting a pull request
+
+If you are a first time contributor to OpenJFX, welcome! Please do the following before you submit your first pull request:
+
+* Sign the Contributor Agreement
+
+    In order for us to evaluate your contribution, you need to sign the [Oracle Contributor Agreement](http://www.oracle.com/technetwork/community/oca-486395.html) (OCA). Ultimately, the goal is to send accepted Pull Requests upstream to the OpenJFX repository hosted on openjdk.java.net. We are not asking you to give up your copyright, but to give us the right to distribute your code without restriction. By doing this you assert that the code you contribute is *yours* to contribute, and not third-party code that you do not own. We ask this of all contributors in order to assure our users of the origin and continuing existence of the code. You only need to sign the OCA once.
+
+* Read the code review policies
+
+    Please read the entire section below on how to submit a pull request, as well as the [OpenJFX Code Review Policies](https://wiki.openjdk.java.net/display/OpenJFX/Code+Reviews). If this is a feature request, please note the additional requirements and expectations in the [New features / API additions](https://wiki.openjdk.java.net/display/OpenJFX/Code+Reviews#CodeReviews-NewFeaturesC.Newfeatures/APIadditions.) section of the Code Review Policies doc.
+
+TIP: A GitHub pull request (PR) should not be the first time we hear about your proposed change to OpenJFX. Unless clearly identified as experimental or work-in-progress (WIP), we will usually close a pull request that isn't associated with an existing bug report. Reading the policies below will help you in getting your change approved.
+
+### Submitting your changes via a pull request
 
 Once your changes and tests are ready to submit for review:
 
 1. Test your changes
 
-    Run the test suite to make sure that nothing is broken. 
+    Run the test suite to make sure that nothing is broken.
 
-2. Sign the Contributor License Agreement 
+2. Rebase your changes
 
-    Ultimately, the goal is to send accepted Pull Requests upstream to the OpenJFX repository hosted at Oracle infrastructure. Therefore, please make sure you have signed the [Oracle Contributor Agreement](http://www.oracle.com/technetwork/community/oca-486395.html) . We are not asking you to give up your copyright, but to give us the right to distribute your code without restriction. We ask this of all contributors in order to assure our users of the origin and continuing existence of the code. You only need to sign the OCA once.
+    Update your local repository with the most recent code from the main [repository]((https://github.com/javafxports/openjdk-jfx)), and rebase your branch on top of the latest `develop` branch. We prefer your initial changes to be squashed into a single commit. Later, if we ask you to make changes, add them as separate commits. This makes them easier to review.
```

Comment:
we can probably add a link to a general github documentation page that explains the rebase command?

Reply:
Good idea.